### PR TITLE
use ip as fqdn from load balancer if missing

### PR DESCRIPTION
**Pull request by meiserloh**

### CODE REVIEW

File: app/setup/data/fqdnProducerStep.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+// NewFQDNCreatorStep creates a new setup step sets the FQDN
+func NewFQDNCreatorStep(config *context.SetupConfiguration, clientSet kubernetes.Interface, namespace string) *fqdnCreatorStep {
+	return &fqdnCreatorStep{config: config, clientSet: clientSet, namespace: namespace}
+}
+
+// GetStepDescription return the human-readable description of the step
+func (fcs *fqdnCreatorStep) GetStepDescription() string {
+	return fmt.Sprintf("Creating a new FQDN from the IP of a loadbalancer service")
```
Comment (cesmarvin) [inline app/setup/data/fqdnProducerStep.go:46]: 🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1039: unnecessary use of fmt.Sprintf

File: app/setup/data/fqdnProducerStep.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cesLoadbalancerName,
+			Namespace: fcs.namespace,
+			Labels:    map[string]string{"app": "ces"},
+		},
+		Spec: corev1.ServiceSpec{
+			Type:           corev1.ServiceTypeLoadBalancer,
+			IPFamilyPolicy: &ipSingleStackPolicy,
+			IPFamilies:     []corev1.IPFamily{corev1.IPv4Protocol},
+			Selector:       map[string]string{DoguLabelName: "nginx-ingress"},
```
Comment (nhinze23) [inline app/setup/data/fqdnProducerStep.go:87]: Use a contanst for `nginx-ingress`

File: app/setup/data/fqdnProducerStep.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func (fcs *fqdnCreatorStep) setFQDNFromLoadbalancerIP(ctx gocontext.Context) error {
+	return retry.OnError(backoff, serviceRetry, func() error {
+		logrus.Debug("Try retrieving service...")
+		service, err := fcs.clientSet.CoreV1().Services(fcs.namespace).Get(ctx, cesLoadbalancerName, metav1.GetOptions{})
+		if err != nil || len(service.Status.LoadBalancer.Ingress) <= 0 {
+			logrus.Debugf("wait for service %s to be instantiated", cesLoadbalancerName)
+			return fmt.Errorf("service not yet ready %s: %w", cesLoadbalancerName, err)
+		}
```
Comment (nhinze23) [inline app/setup/data/fqdnProducerStep.go:113]: If other errors than "not found" occur this routine will wait and retry unnecassary. 

Wait only if you get a not found error or the LB has no IP. Terminate if an other error occurs.

You can use `errors.IsNotFound(err)` from `"k8s.io/apimachinery/pkg/api/errors"`

File: app/setup/data/fqdnProducerStep.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+		loadbalancerIP := service.Status.LoadBalancer.Ingress[0].IP
+		fcs.config.Naming.Fqdn = loadbalancerIP
+		logrus.Infof("Loadbalancer Ip succesfully retrieved and set as new FQDN")
```
Comment (nhinze23) [inline app/setup/data/fqdnProducerStep.go:117]:
```suggestion
		logrus.Infof("Loadbalancer IP succesfully retrieved and set as new FQDN")
```

File: app/setup/data/fqdnProducerStep.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	return fcs.setFQDNFromLoadbalancerIP(ctx)
+}
+
+func (fcs *fqdnCreatorStep) checkIfNginxWillBeInstalled() error {
+	for _, dogu := range fcs.config.Dogus.Install {
+		if strings.Contains(dogu, "nginx-ingress") {
+			return nil
+		}
+	}
+	// nginx-ingress not found
```
Comment (nhinze23) [inline app/setup/data/fqdnProducerStep.go:71]:
```suggestion
```

File: app/setup/executor.go [hunk 6]
```diff
@@ -180,6 +180,12 @@ func (e *Executor) RegisterDoguInstallationSteps() error {
 	return nil
 }
```
Comment (nhinze23) [inline app/setup/executor.go:182]: Missing comment

File: app/setup/executor_test.go [hunk 7]
```diff
@@ -142,6 +142,24 @@ func TestExecutor_PerformSetup(t *testing.T) {
 	})
 }
 
+func TestExecutor_RegisterFQDNCreatorStep(t *testing.T) {
+	t.Run("successfully register FQDNC step", func(t *testing.T) {
```
Comment (nhinze23) [inline app/setup/executor_test.go:146]:
```suggestion
	t.Run("successfully register FQDN step", func(t *testing.T) {
```

**Pull request merged**